### PR TITLE
Fill mode fixes

### PR DESCRIPTION
**Pull request by weavermedia**
Fixes for issue #375 

- show FILL conditional value instead of "PROBABILITY: 0%" (missed this during the last community merge)
- improve runtime setting for OLED, showing "Sync Scaling" instead of "SCAL"

I don't have the time to fix the latching fill value right now. I looked at the code but don't fully understand the logic behind `prevBase`.

@m-m-adams I added the iteration constant to your note row probability function. Shouldn't have done that in this PR really. Let me know if I should move it to a separate PR.

### CODE REVIEW

**Comment by seangoodvibes** [comment]
@weavermedia please push fixes to release 1.0 branch. Thanks

**Comment by weavermedia** [comment]
@seangoodvibes I don't know what that means or how to do it, sorry.

**Comment by seangoodvibes** [comment]
> @seangoodvibes I don't know what that means or how to do it, sorry.

At the top of your PR it shows what you're merging commits from your branch to community.

Click on community and change it to Release 1.0.

As we are now prepping for our first release, a new branch was created as a copy of the community branch and all fixes are to be made in that branch.

The community branch is only for new features that are not part of the first release.

image

**Comment by weavermedia** [comment]
@seangoodvibes Sorry, I don't have time to dig into this right now. If anyone can take over I'm fine with that.

**Comment by seangoodvibes** [comment]
Also is there duplicate work in here with the work in this PR from @soymonitus 

https://github.com/SynthstromAudible/DelugeFirmware/pull/384

**Comment by m-m-adams** [comment]
Yeah but it doesn't matter, they're the same change so no delta

**Review by m-m-adams — APPROVED** [review]

File: src/deluge/gui/views/instrument_clip_view.cpp [hunk 1]
```diff
@@ -2839,7 +2839,8 @@ void InstrumentClipView::setRowProbability(int32_t offset) {
 	uint8_t probabilityValue = noteRow->probabilityValue;
 	bool prevBase = false;
 	// Covers the probabilities and iterations
-	probabilityValue = std::clamp<int32_t>((int32_t)probabilityValue + offset, (int32_t)0, kNumProbabilityValues + 35);
+	probabilityValue = std::clamp<int32_t>((int32_t)probabilityValue + offset, (int32_t)1,
```
Comment (m-m-adams) [inline src/deluge/gui/views/instrument_clip_view.cpp:2842]: This needs to start from 0 to accomodate fill

**Review by m-m-adams — CHANGES_REQUESTED** [review]

**Comment by soymonitus** [comment]
Given that this hasn't been merged yet, is there time to include a fix for this in this same PR?

https://github.com/SynthstromAudible/DelugeFirmware/issues/412

**Comment by trappar** [comment]
This seems to be hiding the fill latching issue in a way that will make it very hard to detect (this is a bad thing). When this it would have happened before and said "0% LATCHING" it now says "FILL", so in this case there are two entries in the probability list that say "FILL". Super hard to detect since it can appear to simply be a missed input. I don't think this should be merged in this state. I'd prefer it say "FILL LATCHING" even if that option shouldn't exist so that it's easier to fix in a future PR.

**Comment by soymonitus** [comment]
I agree with @trappar in both things.
Fill latching should have its own name so we can address it in the future and so the user knows that only Fill (without latching) works
And agree to address my issue in a different PR

File: src/deluge/gui/views/instrument_clip_view.cpp (hunk 1 above)
Comment (weavermedia) [reply src/deluge/gui/views/instrument_clip_view.cpp]: Fixed in https://github.com/SynthstromAudible/DelugeFirmware/pull/399/commits/22de11a907d187964e282964d95c2bb7e12dc217

**Comment by weavermedia** [comment]
@trappar This fix wasn't meant to touch the latching behavior. Unfortunately I don't have time to look dig into it right now. I looked at the prevBase behavior but couldn't figure it out. Happy for someone else to jump in and resolve.

**Comment by weavermedia** [comment]
@soymonitus Fill latching shouldn't exist at all. It's something to do with prevBase but I don't have time to dig into it right now.

And agreed, the note repeat issue should stay separate.

**Review by m-m-adams — APPROVED** [review]

**Comment by m-m-adams** [comment]
> This seems to be hiding the fill latching issue in a way that will make it very hard to detect (this is a bad thing). When this it would have happened before and said "0% LATCHING" it now says "FILL", so in this case there are two entries in the probability list that say "FILL". Super hard to detect since it can appear to simply be a missed input. I don't think this should be merged in this state. I'd prefer it say "FILL LATCHING" even if that option shouldn't exist so that it's easier to fix in a future PR.
> 
> And @soymonitus it's probably easier if each issue can be addressed in an individual PR to keep the footprint small.

I think it's good to merge - while displaying fill for both is non-ideal I think it's better than displaying FILL latching when that doesn't exist. The prev base logic is complex and modifying it is better in a separate PR

**Comment by m-m-adams** [comment]
Also on 7seg this still sets the latching dot anyway

**Comment by sichtbeton** [comment]
Is there a setting  to mute a note when pressing “fill / sync scale”?

**Comment by soymonitus** [comment]
> Is there a setting  to mute a note when pressing “fill / sync scale”?

This is known as Not-Fill but it is not implemented yet. It will be in the future

**Pull request merged**